hudi-common/src/main/java/org/apache/hudi/common/table/timeline/LSMTimeline.java

```diff
@@ -142,7 +142,17 @@ public static Schema getReadSchema(HoodieArchivedTimeline.LoadMode loadMode) {
   public static boolean isFileInRange(HoodieArchivedTimeline.TimeRangeFilter filter, String fileName) {
     String minInstant = getMinInstantTime(fileName);
     String maxInstant = getMaxInstantTime(fileName);
-    return filter.isInRange(minInstant) || filter.isInRange(maxInstant);
+    return  // min instant hits the range
+        filter.isInRange(minInstant)
+            ||
+            // max instant hits the range
+            filter.isInRange(maxInstant)
+            ||
+            // the [min, max] covers the whole filter range, filter's start/end can't be null here,
+            // because min and max are a finite set. We hope that the boundary of the filter is within this finite set, so it naturally cannot be infinite
+            (filter.startTs != null && InstantComparison.compareTimestamps(filter.startTs, InstantComparison.GREATER_THAN_OR_EQUALS, minInstant)
```

Comment:
Here, we only need to consider that the boundary of the filter is within the range of [min, max], and both min and max are finite sets. Therefore, if the filter is to be within [min, max], neither startTs nor endTs can be null, because in the current implementation,null represents infinity. Infinity cannot exist within the range of finite sets

Reply:
> Here, we only need to consider that the boundary of the filter is within the range of [min, max], and both min and max are finite sets. Therefore, if the filter is to be within [min, max], neither startTs nor endTs can be null, because in the current implementation,null represents infinity. Infinity cannot exist within the range of finite sets

This part requires considering this kind of infinity and various boundary cases outside the `TimRangeFilter` class, which is rather troublesome. That's why I proposed pr https://github.com/apache/hudi/pull/14366 for refactoring `TimRangeFilter`